arbnode/inbox_reader.go

```diff
@@ -633,16 +638,26 @@ func (r *InboxReader) run(ctx context.Context, hadError bool) error {
 }
 
 func (r *InboxReader) addMessages(ctx context.Context, sequencerBatches []*SequencerInboxBatch, delayedMessages []*DelayedInboxMessage) (bool, error) {
-	err := r.tracker.AddDelayedMessages(delayedMessages)
+	batch, prevMessageCount, err := r.tracker.AddDelayedMessages(delayedMessages)
 	if err != nil {
 		return false, err
 	}
+
 	err = r.tracker.AddSequencerBatches(ctx, r.client, sequencerBatches)
 	if errors.Is(err, delayedMessagesMismatch) {
 		return true, nil
 	} else if err != nil {
 		return false, err
 	}
+
+	// Only stream batches if there are batches to stream
+	if batch != nil {
+		err = r.tracker.ReorgAtAndEndBatch(batch, prevMessageCount)
+		if err != nil {
+			return false, err
+		}
+	}
```

Comment:
Now that we have separated batch streaming from `AddDelayedMessages`, were there any state assumptions that needs to happen between `AddDelayedMessages` and `AddSequencerBatches`. A good test would be to have a node fail in between those calls and attempting to recover